iceoryx_utils/test/moduletests/test_posix_mutex.cpp

```diff
@@ -49,28 +44,43 @@ class Mutex_test : public Test
         }
     }
 
-    iox::posix::mutex sut{false};
+    iox::posix::mutex sutNonRecursive{false};
+    iox::posix::mutex sutRecursive{true};
+    iox::units::Duration m_watchdogTimeout = 5_s;
+    Watchdog m_deadlockWatchdog{m_watchdogTimeout};
 };
 
 TEST_F(Mutex_test, TryLockWithNoLock)
 {
-    EXPECT_THAT(sut.try_lock(), Eq(true));
-    EXPECT_THAT(sut.unlock(), Eq(true));
+    EXPECT_THAT(sutNonRecursive.try_lock(), Eq(true));
+    EXPECT_THAT(sutNonRecursive.unlock(), Eq(true));
 }
 
-TEST_F(Mutex_test, TryLockWithLock)
+
+#ifndef _WIN32
+TEST_F(Mutex_test, TryLockWithNonRecursiveLock)
```

Comment:
Could we be a bit more explicit with all the test names in this file? I'm kind of missing the expected result for each test, e.g. `TryLockAfterNonRecursiveLockReturnsFalse` or so. Same for most other tests here.

Reply:
Correct, i made them more expressive.